src/main/java/com/example/resourcemanagerapp/controller/MyResourceController.java

```diff
@@ -0,0 +1,85 @@
+package com.example.resourcemanagerapp.controller;
+
+
+import com.example.resourcemanagerapp.additionalTypes.EnumChecker;
+import com.example.resourcemanagerapp.additionalTypes.MyResourceType;
+import com.example.resourcemanagerapp.service.MyResourceServiceImpl;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.configurationprocessor.json.JSONException;
+import org.springframework.boot.configurationprocessor.json.JSONObject;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+public class MyResourceController {
+
+    private final MyResourceServiceImpl myResourceService;
+
+    @Autowired
+    public MyResourceController(MyResourceServiceImpl myResourceService) {
+        this.myResourceService = myResourceService;
+    }
+
+    @PostMapping(value = "/resource/add")
+    public ResponseEntity addResource(@RequestParam String name, @RequestParam Integer userId, @RequestParam String type,
+                                      @RequestParam String metadata){
+        if(name.length() == 0 || userId <= 0 || type.length() == 0 || metadata.length() == 0 ||
+                !checkName(name))
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("Invalid params");
+        MyResourceType myResourceType = EnumChecker.containsResourceType(type);
+        if( myResourceType == null)
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("Invalid params");
```

Comment:
pisalbym nawiasy nawet tam gdzie to nie jest konieczne, wyeliminuje to potencjalne bledy w przyszlosci